src/Compilers/CSharp/Test/Semantic/Semantics/BindingTests.cs

```diff
@@ -3420,27 +3420,72 @@ public void MissingTypeArgumentInGenericExtensionMethod()
 @"
 public static class FooExtensions
 {
-    public static T Foo<T>(this object obj) => default(T);
+    public static object ExtensionMethod0(this object obj) => default(object);
+    public static T ExtensionMethod1<T>(this object obj) => default(T);
+    public static T1 ExtensionMethod2<T1, T2>(this object obj) => default(T1);
 }
 
 public class Class1
 {
     public void Test()
     {
-        var defaultA = ""a"".Foo<>();
-        var defaultB = FooExtensions.Foo<>(""b"");
+        var omittedArg0 = ""string literal"".ExtensionMethod0<>();
+        var omittedArg1 = ""string literal"".ExtensionMethod1<>();
+        var omittedArg2 = ""string literal"".ExtensionMethod2<>();
+
+        var moreArgs0 = ""string literal"".ExtensionMethod0<int>();
+        var moreArgs1 = ""string literal"".ExtensionMethod1<int, bool>();
+        var moreArgs2 = ""string literal"".ExtensionMethod2<int, bool, string>();
+
+        var lessArgs1 = ""string literal"".ExtenionMethod1();
+        var lessArgs2 = ""string literal"".ExtenionMethod2<int>();
+
+        var nonExistingMethod0 = ""string literal"".ExtensionMethodNotFound0();
+        var nonExistingMethod1 = ""string literal"".ExtensionMethodNotFound1<int>();
+        var nonExistingMethod2 = ""string literal"".ExtensionMethodNotFound2<int, string>();
+
+        var exactArgs0 = ""string literal"".ExtensionMethod0();
+        var exactArgs1 = ""string literal"".ExtensionMethod1<int>();
+        var exactArgs2 = ""string literal"".ExtensionMethod2<int, bool>();
     }
 }
 ";
             var compilation = CreateCompilationWithMscorlibAndSystemCore(source);
 
             compilation.VerifyDiagnostics(
-                // (11,24): error CS0305: Using the generic method group 'Foo' requires 1 type arguments
```

Comment:
>// (11,24): error CS0305: Using the generic method group 'Foo' requires 1 type arguments [](start = 16, length = 88)

I think we should keep this error. My recommendation would be to tot change implementation and simply add more tests.